composer.json

```diff
@@ -50,7 +50,6 @@
         "friendsofsymfony/jsrouting-bundle": "^3.0",
         "friendsofsymfony/rest-bundle": "^3.2",
         "gedmo/doctrine-extensions": "^3.8",
-        "guzzlehttp/promises": "^1.0 || ^2.0",
```

Comment:
the `promises` package need to stay will not be removed they are required that the http cache working performant with the symfony http client: https://foshttpcachebundle.readthedocs.io/en/stable/overview.html#installation

Reply:
Interesting. Very weird that this is not a dependency of either of those packages. Because we don't use any of the classes in the Sulu source code.

Reply:
@mamazu you can use any php http implementation with the FosHttpCache component.

https://github.com/FriendsOfSymfony/FOSHttpCache/blob/6151c80350a1c54e8723f927a50ee2f90fcd19f9/composer.json#L29

As Symfony based we choose the Symfony Http Client it provides the async: 

https://github.com/symfony/http-client/blob/955e43336aff03df1e8a8e17daefabb0127a313b/composer.json#L19

But its async implementation depends on guzzle promises which is not a direct dependency as you can use it without async still.

We may can discuss with @dbu if the FOSHttpCacheBundle as it is a symfony bundle may could require it as I think its uncommon that somebody using another client inside a symfony project?

Reply:
Yeah I think that would make sense. But that's probably a topic for another pull request.

Reply:
FOSHttpCache [uses](https://github.com/FriendsOfSymfony/FOSHttpCache/blob/6151c80350a1c54e8723f927a50ee2f90fcd19f9/composer.json#L28) [php-http/discovery](https://docs.php-http.org/en/latest/discovery.html) which contains a composer plugin that requires a [suiteable http client](https://docs.php-http.org/en/latest/discovery.html#auto-installation) if none is already available in your project.